File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.cpp
class Solution {
public:
    int countCollisions(string directions) {
        int res = 0;
        bool stationary = false;
        for(int i = 0; i < directions.size(); i++)
        {
            if(directions[i] == 'R')
            {
                int count = 0;
                while(i < directions.size() && directions[i] == 'R')
                {
                    count++;
                    i++;
                }
                if(i < directions.size())
                {
                    if(directions[i] == 'S')
                    {
                        res += count;
                    }
                    else
                    {
                        res += count + 1;
                    }
                    stationary = true;
                }
            }
            else if(directions[i] == 'S')
            {
                stationary = true;
            }
            else
            {
                if(stationary)
                {
                    res++;
                }
            }
        }
        return res;
    }
};

File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        res = 0
        stationary = False
        countR = 0
        for d in directions:
            if d == 'R':
                countR += 1
            elif d == 'S':
                if countR > 0:
                    res += countR
                    countR = 0
                stationary = True
            else:
                if countR > 0:
                    res += countR + 1
                    countR = 0
                    stationary = True
                elif stationary:
                    res += 1
        return res
